node/main-extract-async.ts

```diff
@@ -107,47 +108,27 @@ export function sendNewVideoMetadata(imageElement: ImageElement) {
  */
 export function metadataQueueRunner(file: TempMetadataQueueObject, callback) {
 
-  console.log('checking metadata for', file.fullPath);
-
-  let found = false;
+  console.log('metadata check:', file.fullPath);
 
   if (!file.generateHashes) {
 
-    for (let i = 0; i < cachedFinalArray.length; i++) {
-      const element = cachedFinalArray[i];
-      if (element.partialPath === file.partialPath && element.fileName === file.name) { // consider comparing full path in case source folders overlap
-        found = true;
-        break;
-      }
-    }
-    console.log('found:', found);
-
-    if (found) {
+    if (cachedMetaExtracted.has(file.fullPath)) {
```

Comment:
@cal2195 -- thought you might like this minor rework 😉 

Rather than search through the array to see if `fullPath` matches I created a _map_ so it's just a matter of lookup `map.has(string)` 😎 

All this is still WIP -- but I wanted to share 👋

Reply:
How did we not think of this?! 😅 

Although I must say, VHA is one of the most performant applications I've ever used for the amount of data it processes on the fly! I wonder how much optimisation you get with pipes for free... 💭

Reply:
Yeah -- Angular pipes only re-render if any of their inputs change - so it's not the whole stack of pipes that runs when you change something - and many pipes just return whatever comes into them. And since it's not literally "loading" the data into the pipe - but just receiving a reference to a location, it can pass it along to the next pipe as if it was a single integer - fast 🚀 😎

Reply:
Also -- after looking at the code some more I couldn't figure out why we were ever looking whether a hash was present in previously-added files, so I removed this piece of logic.

Please let me know if I'm missing something / going to run into a case where it's needed later 😅

Reply:
I named one variable `alreadyInAngular` (for transparency about its function) -- and it gets populated with full paths to all the files in the currently-opened vha file. 

When a watcher finds a new file, it compares the full path to the list -- if it's on the list, we are done. If it's _not_ on the list, we `createNewElement` (and create a _hash_ for it of course).

When we send over the newly-created element to Angular, we update the `alreadyInAngular` with full path -- so if a subsequent watcher finds the same file it doesn't create a copy.

🤔  -- seems like it should work out fine.

🙇 thank you again for the amazing setup -- all I'm doing now is just cleaning up a few things and making it work with more than one input folder -- the functionality is all yours 🤝 🎉 🙌